### ISSUE TITLE
error on line 87

### ISSUE THREAD

**Issue by hugeblank** [eric-wieser/computercraft-github#21]
After cloning so many times I get this error message:
`github:87: Could not get github API from repos/hugeblank/BagelBot/git/trees/unstable-as`
What would be the cause of this? the repo is in existence, and I had no prior problems with it. I have also found that after a 5-10 minute wait that this issue fixes itself, but it eventually occurs again.

**Comment by eric-wieser** [comment]
You're hitting the GitHub rate limiting. It would be good if I could give a better error message, but from what I remember the http functions don't give you the status code.

Authenticating should up your allowed rate

**Comment by hugeblank** [comment]
yeah that would explain it. Thanks for the info, I figured it would have something to do with that.

**Comment by danports** [comment]
Thanks guys - I ran into the same thing and this issue was really helpful. A simple program like this can display your rate limit status (without authentication):

`print(http.get("https://api.github.com/rate_limit", nil).readAll())`

@eric-wieser It might be helpful to modify the text of the error raised to suggest rate limiting as a possible cause and authentication as the solution.

**Comment by danports** [comment]
That's one option...I was thinking something even simpler to start with, i.e. just adding a suggestion to check your rate limit to the string passed to `error`, without adding any new logic.
